### DISCUSSION TITLE
Suspicious code in `hasDiagonalArtifactInner`

### DISCUSSION THREAD

**Discussion by bluebear94** [Chlumsky/msdfgen#190] [Q&A] [answered]
I’m trying to port msdfgen’s error correction algorithm to my pure-Rust implementation [fdsm](https://gitlab.com/Kyarei/fdsm), and I found two potential problems with the `hasDiagonalArtifactInner` function in msdfgen:

* [The linear term passed to `solveQuadratic`](https://github.com/Chlumsky/msdfgen/blob/master/core/MSDFErrorCorrection.cpp#L303) subtracts `dA` twice. I think `dBC - 2.0 * dA` is what was meant here, but it would be good to at least leave a comment that subtracting by `dA` twice is intentional and not a typo.
* When checking `xm` against the interpolated medians at the local extrema, [the `em` variable is never used](https://github.com/Chlumsky/msdfgen/blob/master/core/MSDFErrorCorrection.cpp#L317).

**Comment by Chlumsky** [comment, marked as answer]
Thanks for finding this. You are right that `em` not being used is an error. I have committed a fix. `dBC-dA-dA` is equivalent to `dBC-2*dA`, yes. If it helps you in any way I might be able to provide you with the "testbed" I originally developed this with.
